dmscripts/send_dos_opportunities_email.py

```diff
@@ -67,47 +95,122 @@ def get_html_content(briefs, number_of_days):
+    period between DOS framework iterations, we need to support both iterations.
+
+    For example, on the second day of DOS4, any DOS3 briefs created the first day (before the go-live switchover) will
+    still need to be emailed to DOS3 suppliers, as well as the usual DOS4 briefs sent to DOS4 suppliers.
+    """
     start_date = date.today() - timedelta(days=number_of_days)
     end_date = date.today() - timedelta(days=1)
 
-    live_briefs = get_live_briefs_between_two_dates(
-        data_api_client, lot_data["lot_slug"], start_date, end_date, framework_slug
-    )
-    if not live_briefs:
+    # Get all live briefs for each DOS framework on the lot (or a single DOS framework, if supplied as a script arg)
+    live_briefs_by_framework = get_live_briefs_by_framework_and_lot(data_api_client, start_date, end_date)
+
+    # If specific framework script arg supplied, ignore other frameworks
+    if framework_override:
+        if not live_briefs_by_framework.get(framework_override):
+            logger.info(
+                "No new briefs found for {} in the last {} day(s)".format(framework_override, number_of_days)
+            )
+            return True
+        live_briefs_by_framework = {framework_override: live_briefs_by_framework.get(framework_override)}
+
+    # If no briefs found, exit early
+    if not live_briefs_by_framework.keys():
         logger.info(
-            "No new briefs found for '{0}' lot".format(lot_data["lot_slug"]),
+            "No new briefs found for DOS frameworks in the last {} day(s)".format(number_of_days),
             extra={"number_of_days": number_of_days}
         )
         return True
-    logger.info(
-        "{0} new briefs found for '{1}' lot".format(len(live_briefs), lot_data["lot_slug"])
-    )
-
-    campaign_data = get_campaign_data(lot_data["lot_name"], lot_data["list_id"], live_briefs[0]['frameworkName'])
-    logger.info(
-        "Creating campaign for '{0}' lot".format(lot_data["lot_slug"])
-    )
-    campaign_id = mailchimp_client.create_campaign(campaign_data)
-    if not campaign_id:
-        return False
-
-    content_data = get_html_content(live_briefs, number_of_days)
-    logger.info(
-        "Setting campaign data for '{0}' lot and '{1}' campaign id".format(lot_data["lot_slug"], campaign_id)
-    )
-    if not mailchimp_client.set_campaign_content(campaign_id, content_data):
-        return False
-
-    logger.info(
-        "Sending campaign for '{0}' lot and '{1}' campaign id".format(lot_data["lot_slug"], campaign_id)
-    )
-    if not mailchimp_client.send_campaign(campaign_id):
-        return False
+
+    for framework_slug in live_briefs_by_framework.keys():
+        for lot_slug, live_briefs in live_briefs_by_framework[framework_slug].items():
+
+            if lot_slug_override and lot_slug != lot_slug_override:
+                logger.warning("Skipping campaign for '{0}' lot on {1}".format(lot_slug, framework_slug))
```

Comment:
Why do these braces have numbers in while the braces on 162 do not?

Reply:
Haha, probably because I refactored them on different days 😹Will make them consistent